### ISSUE TITLE
Conditional return type for Ds\Map::get() still not working

### ISSUE THREAD

**Issue by enumag** [vimeo/psalm#3453]
Related to #3405 and #3423

```php
<?php declare(strict_types = 1);

namespace App;

use DateTimeImmutable;
use Ds\Map;

abstract class Z
{
	public function test(): void
	{
		$map = $this->createMap();

		$date = $map->get('test');

		echo $date->format('Y');
	}

	/**
	 * @return Map<string, DateTimeImmutable>
	 */
	abstract protected function createMap(): Map;
}
```

This code still causes a false-positive (using dev-master):

```
ERROR: PossiblyNullReference - src/Z.php:16:15 - Cannot call method format on possibly null value (see https://psalm.dev/083)
                echo $date->format('Y');
```

**Comment by muglug** [comment]
Sorry, conditional returns were not working on instance methods – now they are.

**Comment by enumag** [comment]
@muglug I'm afraid there must be some other issue. The code I posted in this issue still causes false positive even after I updated to the latest commit.

**Comment by muglug** [comment]
Ah sorry, that's because my fix was just for user-defined methods.

**Comment by enumag** [comment]
Awesome. I'll try again tomorrow. Thanks for making this work!
